packages/core/src/router/metadata.ts

```diff
@@ -167,3 +169,103 @@ export function operation(verb: string, path: string, spec: OperationObject) {
     );
   };
 }
+
+/**
+ * Describe an input parameter of a Controller method.
+ *
+ * @param paramSpec Parameter specification.
+ */
+export function param(paramSpec: ParameterObject) {
+  return function(
+    target: any,
+    propertyKey: string,
+    descriptor: PropertyDescriptor,
+  ) {
+    assert(
+      typeof target[propertyKey] === 'function',
+      '@param decorator can be applied to methods only',
+    );
+
+    let operationSpec: OperationObject = Reflector.getMetadata(
+      'loopback:operation-spec',
+      target,
+      propertyKey,
+    );
+
+    if (!operationSpec) {
+      operationSpec = {
+        responses: {},
+      };
+    }
+
+    if (!operationSpec.parameters) {
+      operationSpec.parameters = [];
+    }
+
+    operationSpec.parameters.push(paramSpec);
+
+    Reflector.defineMetadata(
+      'loopback:operation-spec',
+      operationSpec,
+      target,
+      propertyKey,
+    );
+  };
+}
+
+// TODO(bajtos) @param.IN.TYPE('foo', {required: true})
+export namespace param {
```

Comment:
> Does this override the previous `export function param`

TypeScript merges these two declarations into a single one. See https://stackoverflow.com/a/33392782/69868 and http://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-namespaces.

You cannot write `param.query = {...}`, because a function does not have any `query` property. So the TypeScript way is to define a namespace with the same name as the function has, and define extra properties in that namespace.